### ISSUE TITLE
Transcribe the octal listing

### ISSUE THREAD

**Issue by thewonderidiot** [virtualagc/virtualagc#371]
Transcribe the octal listing, which ranges from pages 1590 through 1733. Probably using an updated octopus to run things through tesseract, with some manual transcription on pages that are kinda messed up.

**Comment by thewonderidiot** [comment]
@rburkey2005, I've found a tweak to octopus that produces very, very clear digits for Luminary 210. Replacing [this line](https://github.com/virtualagc/virtualagc/blob/ff1d8e5ce339ad1623f5822ef6a4b3d046e16e5e/Tools/ocr/octopus.py#L45), which performs the thresholding, with a wide-area adaptive mean threshold:

` thresh = cv2.adaptiveThreshold(blurred, 255, cv2.ADAPTIVE_THRESH_MEAN_C, cv2.THRESH_BINARY, 201, 11)

seems to work quite well. It does look like it's crashing on some of the pages though, so I need to look into that.

**Comment by rburkey2005** [comment]
Okay, thanks.  

I'm thinking that in terms of bootstrapping this process, at first I need to go through the following steps:

1. Manually input first memory bank, and completely proof/debug it.
2. Use already-converted blocks to train the OCR.
3. OCR _next_ memory bank.
4. Completely proof/debug this memory bank, if it works well enough.  (If not, manually input.)
5. Go back to step #2.

It seems like after I've done that a few times, the OCR'ing should progressively improve. Once the error rate during proofing seems fairly low, I could then afford to OCR a few (or a lot) of memory banks before needing to retrain.

So at first, I'll only need to be OCR'ing the first few memory banks.

**Comment by thewonderidiot** [comment]
Excellent, sounds like a plan. Just that change should get you through the first few pages well enough. While you work on that, I'll focus on less important tweaks and manually processing the images that octopus is choking on.

**Comment by thewonderidiot** [comment]
@rburkey2005, I've finished with image preprocessing. Here's a zip file containing all of the images, ready for OCR: [lum210_prepared.zip](https://github.com/virtualagc/virtualagc/files/599198/lum210_prepared.zip)

There's two remaining problem pages that will have to be dealt with during transcription: 1702 (which is very dirty on the bottom right) and 1724 (which has the line overprinting). Outside of that, the images look mostly clean, and should work fine for tesseract. I added the additional step of cropping out extra whitespace on partial pages, so hopefully that'll be one less thing to throw us off this time.

**Comment by rburkey2005** [comment]
Okay. I haven't gotten started with any of it yet, because I'm trying to finish some unrelated issues, so perhaps tomorrow I can finally take a look.

**Comment by rburkey2005** [comment]
@thewonderidiot Yes, those are astoundingly clear.  I'm not sure I would believe it if I didn't see it.

**Comment by rburkey2005** [comment]
@thewonderidiot In spite of my promises, I didn't start working with it until today.  The TIFFs were so clear looking that on a whim I stuck them into OmniPage rather than Tesseract (since it was a lot easier for me, because there's nothing extra for me to install and no training involved), and the OCR results are quite good.  I've fixed all the issues reported by oct2bin, but haven't done any other proofing as such.  But even so, 10 of the 36 bank checksums are already good.  I think that's a really good sign, so I'm going to stick with OmniPage's OCR for now.

Regarding the issue of pages that crashed for you, there were 6 pages that OmniPage refused to load, claiming that their DPI was too low.  Perhaps the same thing was happening to you.  The first one, I think, was p. 1622.  Anyway, I "fixed" it simply by converting all of the TIFFs to PNGs.  Oddly, OmniPage also insisted that 4-5 (other) pages were upside-down, which they weren't.  :-)

**Comment by thewonderidiot** [comment]
@rburkey Awesome!! I guess training does make less sense when all of the numbers actually look like they're supposed to. :) 10 out of 36 banks with correct checksums is definitely a good sign!

I think OmniPage's gripes are unrelated to my Octopus crashings. All of those were cases of pages for which the hole punches down the left side hadn't been cropped off. Octopus was treating those as significant, and then at some point in the cropping process at the end, it was ending up with a 0x0 image.

In addition to the four or five of those that I had to pre-crop before feeding into Octopus, there were also a handful that I had to clean up manually in GIMP afterwards. 1622 looks familiar -- it has a long pen or pencil mark crossing through a 0 in the lower right. That was big and dark enough that some of it made it through, so I had to remove it manually. I wonder if the format GIMP saved the TIFFs in was somehow slightly different than what OpenCV produced.

I've got nothing no the upside down pages though. :)

**Comment by rburkey2005** [comment]
Well, it was just a thought, but I'm not too surprised if the issues were actually different ones.

The GIMP does seem to like to set the DPI to 75, and that's my recollection of what OmniPage's complaint was.

**Comment by rburkey2005** [comment]
@thewonderidiot Actually, I had miscounted, and really 12/36 banks had the correct checksums.  

But on a separate though related topic:  I've done a little proofing since then, both with the color images and your TIFFs, and I find it much easier to use the latter, since they're so much easier to read.  I wonder if it mightn't be helpful for the folks doing the source transcribing if we processed _all_ of the pages (not just the octals), and presented those in place of the crummified color images now at ibiblio.  They wouldn't have to be cropped or anything, just color-processed.  And converted to PNG (since the PNGs are smaller than the TIFFs, and a lot of browsers won't display TIFFs without extra plug-ins).

Opinions?

**Comment by thewonderidiot** [comment]
I worry that things like pencil/marker markings will either be lost, or overpower the characters if they intersect. The color images are probably still worth having around if only to help resolve such cases. But it's a good idea, so I've added the --no-crop option to Octopus, and have it processing all of the page images now. So far they're looking pretty good. I'll send them over to you when it finishes.

**Comment by thewonderidiot** [comment]
@rburkey2005: Here's the complete set of processed images: https://dl.dropboxusercontent.com/u/22056193/lum210_bw_pngs.7z

I also short-circuited the despeckling step in Octopus, for fear that it might go on a rampage and destroy meaningful data, so as a whole they're slightly less clean than the prepared octals, but not too much so.

**Comment by rburkey2005** [comment]
@thewonderidiot I did a little more processing (getting rid of the newsletter stuck in the middle, fixing page numbers, reducing the resolution) and uploaded the images to ibiblio.  But I took your advice and kept the old images too.  [The new ones are at](http://www.ibiblio.org/apollo/ScansForConversion/Luminary210-BW/).  I'll let you handle how (or if) you want to communicate that info to the volunteers.  As for myself, though, I guarantee I'll be using the B&W ones.

**Comment by rburkey2005** [comment]
@thewonderidiot I said before that the Luminary 210 octals initially had 10/36 checksums correct after oct2bin-identified errors had been corrected; I miscounted, and it was 12/36.

For Luminary 069, in comparison, 18/36 banks had correct checksums at that point, and the incorrect banks had simple enough errors that I was able to fix thethem in short order.  So even though I have not yet proofed a single page, all 36 banks now have correct checksums. And my _guess_ is that there probably aren't any errors left to be found, because the possible errors left by the OCR'ing process aren't very prone to creating equal and opposite pairs that leave the checksum unchanged ... but time will tell.

In short, this is a pretty sweet method now.
